File: backend/src/main/java/com/collabcloud/model/Comment.java
package com.collabcloud.model;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "comments")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long commentID;

    @Column(columnDefinition = "TEXT")
    private String content;
    private String email;
    private LocalDate createdDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "file_id")
    private FileEntity file;

    public Comment() {
    }

    // getters and setters
    public Long getCommentID() {
        return commentID;
    }

    public void setCommentID(Long commentID) {
        this.commentID = commentID;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public LocalDate getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(LocalDate createdDate) {
        this.createdDate = createdDate;
    }

    public FileEntity getFile() {
        return file;
    }

    public void setFile(FileEntity file) {
        this.file = file;
    }
}

File: frontend/src/models/Comment.ts
// Comment model based on CollabCloud Class Diagram
export interface Comment {
    commentID: string
    content: string
    timestamp: Date
}

export class CommentModel implements Comment {
    commentID: string
    content: string
    timestamp: Date
    email: string
    author: string

    constructor(commentID: string, content: string, author: string, email: string = '') {
        this.commentID = commentID
        this.content = content
        this.timestamp = new Date()
        this.author = author
        this.email = email
    }

    getCommentID(): string {
        return this.commentID
    }

    setCommentID(id: string): void {
        this.commentID = id
    }

    getAuthor(): string {
        return this.author
    }

    setAuthor(author: string): void {
        this.author = author
    }

    getEmail(): string {
        return this.email
    }

    setEmail(email: string): void {
        this.email = email
    }

    getContent(): string {
        return this.content
    }

    setContent(content: string): void {
        this.content = content
    }

    getTimestamp(): Date {
        return this.timestamp
    }

    setTimestamp(date: Date): void {
        this.timestamp = date
    }
}
